numba_cuda/numba/cuda/compiler.py

```diff
@@ -977,6 +978,173 @@ def define_error_gv(postfix):
     return helper_func
 
 
+def compile_all(
+    pyfunc,
+    sig,
+    debug=None,
+    lineinfo=False,
+    device=True,
+    fastmath=False,
+    cc=None,
+    opt=None,
+    abi="c",
+    abi_info=None,
+    output="ltoir",
+    forceinline=False,
+    launch_bounds=None,
+):
+    """Return a list of PTX/LTO-IR for kernel as well as external functions depended by the kernel.
+    If external functions are cuda c/c++ source, they will be compiled with NVRTC. The output code
+    kind is the same as the `output` parameter.
+    Otherwise, they will be passed through to the return list.
+    """
```

Comment:
Because this docstring will appear in the docs, I think it needs some modification to fit into the context of the documentation on the compilation APIs:

```suggestion
    """Similar to ``compile()``, but returns a list of PTX codes/LTO-IRs for
    the compiled function and the external functions it depends on.
    If external functions are CUDA C++ source, they will be compiled with
    NVRTC. Other kinds of external function code (e.g. cubins, fatbins, etc.)
    will be added directly to the return list. The output code kind is
    determined by the ``output`` parameter, and defaults to ``"ltoir"``.
    """
```

This change is aimed at pointing out the differences between `compile()` and `compile_all()`.

The docs change to include this is also required - this could be done by adding

```rst
.. autofunction:: numba.cuda.compile_all
```

around line 126 of `docs/source/reference/host.rst`.